index.js

```diff
@@ -40,9 +39,13 @@ EncryptedField.prototype.vault = function(name) {
 
             var cipher = crypto.createCipheriv(self._algorithm, self.key, new_iv);
 
-            cipher.end(JSON.stringify(value), 'utf-8');
-            var enc_final = Buffer.concat([new_iv, cipher.read()]);
-            var previous = this.setDataValue(name, enc_final);
+            var enc_final = Buffer.concat([
+                cipher.update(new_iv),
```

Comment:
Yeah, that doesn't seem right. Assume there must have been something wrong with my test, will have another look.